### DISCUSSION TITLE
Smart Meter

### DISCUSSION THREAD

**Discussion by speedymk1** [arendst/Tasmota#17507] [Ideas]
Hello,

I got an authentication key and an encyrption key from my engery provider.
This means, that it won't work now with tasmota, I am right?
This means I found no how to for it.

And so I will ask, if there is a plan to change it?
If not, I have to find another solution like an raspberry pi or something else.

**Comment by gemu2015** [comment]
i am currently working on a decryption for sml driver.
there are however several versions with different encryptions. i will not support many of them

look here for a discussion:

https://github.com/arendst/Tasmota/issues/8790

i will have a driver for this kind of meter soon
